numpy/core/records.py

```diff
@@ -847,12 +847,10 @@ def fromstring(datastring, dtype=None, shape=None, offset=0, formats=None,
     return _array
 
 def get_remaining_size(fd):
-    try:
-        fn = fd.fileno()
-    except AttributeError:
-        return os.path.getsize(fd.name) - fd.tell()
-    st = os.fstat(fn)
-    size = st.st_size - fd.tell()
+    pos = fd.tell()
+    fd.seek(0, 2)
+    size = fd.tell() - pos
+    fd.seek(pos, 0)
     return size
```

Comment:
Possibly clearer as:
```suggestion
    pos = fd.tell()
    fd.seek(0, 2)
    try:
        return fd.tell() - pos
    finally:
        fd.seek(pos, 0)
```

I doubt `fd.tell` can fail, but if it does this puts the position back to the right place anyway.

Reply:
in that case maybe move line 850 and 851 under the try ?

Reply:
Agreed, the try-finally block is safer. Fixed accordingly.
@anirudh2290 didn't put line 850 inside the try block, because it defines `pos` and anyways before line 851, we haven't changed the position in the file descriptor so it wont affect.